Formula/gitversion.rb

```diff
@@ -15,16 +15,21 @@ class Gitversion < Formula
   depends_on "dotnet"
 
   def install
-    system "dotnet", "publish",
+    os = "osx"
+    on_linux do
+      os = "linux"
+    end
+
+    system "dotnet", "publish", "src/GitVersion.App/GitVersion.App.csproj",
            "--configuration", "Release",
            "--framework", "net#{Formula["dotnet"].version.major_minor}",
            "--output", libexec,
-           "src/GitVersion.App/GitVersion.App.csproj"
+           "--runtime", "#{os}-x64",
+           "--self-contained", "false",
+           "/p:PublishSingleFile=true"
```

Comment:
Some notes on flags (probably need someone to confirm):
- Previously behavior (excluding these): dotnet produces a framework-dependent, cross-platform (macos + linux/x64 + linux/arm64 + linux/arm + linux/musl + windows + ...) binary in DLL format, which is run via `dotnet <dll>`
- Adding `--runtime <RID>` restricts cross-platform to subset. For example, `--runtime linux-x64` is portable for Linux x64 distros (i.e. not musl, arm, macos, etc). However, it changes default output from framework-dependent to self-contained, which means runtime is included
- Adding `--self-contained false` (or `--no-self-contained`) changes output back to framework-dependent and avoids including runtime.
- Adding `/p:PublishSingleFile=true` (or `-p:PublishSingleFile=true`) consolidates the DLL files into single executable, in this case named `gitversion`.
   - When used in framework-dependent publish, it requires dotnet installation, which is set via `DOTNET_ROOT`.
   - When used in self-contained publish, it works out-of-the-box as runtime is bundled.

The final output seems similar to Java's JAR files where an existing runtime is defined via environment variable, e.g. `DOTNET_ROOT` and `JAVA_HOME`.

I don't know if this is the preferred result though.